modin/engines/base/frame/data.py

```diff
@@ -976,37 +977,80 @@ def _join_index_objects(self, axis, other_index, how, sort):
 
         Parameters
         ----------
-            axis : 0 or 1
-                The axis index object to join (0 - rows, 1 - columns).
-            other_index : Index
-                The other_index to join on.
-            how : {'left', 'right', 'inner', 'outer'}
-                The type of join to join to make.
-            sort : boolean
-                Whether or not to sort the joined index
+        axis : 0 or 1
+            The axis index object to join (0 - rows, 1 - columns).
+        others_index : list(Index)
```

Comment:
As the method is static now, I would prefer renaming `others_list` to something like `indices`.

Reply:
Done.